umcontroller/umcontroller.go

```diff
@@ -174,23 +188,35 @@ const (
 
 const connectionTimeout = 300 * time.Second
 
+const fileScheme = "file"
+
 /***********************************************************************************************************************
  * Public
  **********************************************************************************************************************/
 
 // New creates new update managers controller.
-func New(config *config.Config, storage storage, urlTranslator URLTranslator, certProvider CertificateProvider,
-	cryptcoxontext *cryptutils.CryptoContext, insecure bool) (
+func New(config *config.Config, storage storage, certProvider CertificateProvider,
+	serverContex *cryptutils.CryptoContext, decryptContext Decrypter, insecure bool) (
```

Comment:
Sorry, I made confusion with my multi-proposal comment. In this implementation lets name these params as: `cryptocontext` and `decrypter` accordingly.

Reply:
Done